Add server

typerslow opened the pull request:
Server can handle multiple client connections.

typerslow committed:
server: Initial commit

maxminoS commented:
Unfortunately, due to browsers not supporting raw TCP connections, we will have to use socket.io, a socket implementation for web applications.

Here is an example of an up-to-date framework of socket.io in Go: https://github.com/ambelovsky/gosf

maxminoS commented:
I was wrong.

Apparently you can, using: https://github.com/Rapsssito/react-native-tcp-socket

Hopefully this works, the sockets.io solution was never a great option anyway; the frameworks were heavily outdated.

typerslow committed:
Add message parse + action perform

Types of messages:
`k<key>` Presses a key
`t<text>` Types UTF-8 encoded text
`c<r|m|l>` Clicks right, middle or left mouse button
`p<r|m|l>` Presses right, middle or left mouse button
`r<r|m|l>` Releases right, middle or left mouse button
`m<x> <y>` Moves mouse to relative x,y coords
s<[-]distance> Scrolls
